tools/OutlineService/OutlineService/OutlineService.cs

```diff
@@ -414,67 +441,91 @@ public partial class OutlineService : ServiceBase {
       //    client's call to ResetRouting).
       var interfacesWithIpv4Gateways = NetworkInterface.GetAllNetworkInterfaces()
           .Where(i => i.Name != TAP_DEVICE_NAME)
+          .Where(i => i.OperationalStatus == OperationalStatus.Up)
           .Where(i => i.GetIPProperties().GatewayAddresses
               .Select(g => g.Address)
               .Where(a => a.AddressFamily == AddressFamily.InterNetwork).Count() > 0);
 
       // Ensure there is only one interface with IPv4 gateways.
-      // TODO: When we find multiple interfaces with IPv4 gateways, guess which one is active by,
-      //       for example, choosing the one with the lowest device metric.
       if (interfacesWithIpv4Gateways.Count() < 1) {
-        throw new Exception("no interface has an IPv4 gateway");
+        throw new NoDefautlGatewayFoundException("no interface has an IPv4 gateway");

```

Comment:
Question: what does the user see in the client when they click CONNECT and no network interfaces are up?

Reply:
The connection succeeds when there are no network interfaces during auto-connect, so the app will be in a "CONNECTED" state.

Reply:
Sorry, I mean forget about auto-connect for a minute: if a user clicks CONNECT and there's no network connectivity, what message do they see in the UI? I guess the connectivity checks will fail?

Reply:
That's right, the connectivity checks will fail with a server unreachable error.